### PR TITLE
Mentions Api and Postgres connectivity

### PR DESCRIPTION
**Pull request by houssam-mahlous**
- Setup postgres configuration to run on localhost
- Setup get, post, and delete endpoints in mention controller and connect them with mentions repository
For now, mentions are stored in postgres database but this will be changed to use Solr instead.
- improve code style

### CODE REVIEW

File: common/src/main/java/com/brandwatch/minibcr/common/repository/MentionRepository.java [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
 import com.brandwatch.minibcr.common.domain.Mention;
-import org.springframework.data.jpa.repository.JpaRepository;
-import org.springframework.stereotype.Repository;
 
-@Repository
-public interface MentionRepository extends JpaRepository<Mention, Long> {
+import java.util.List;
+
+public interface MentionRepository{
+    void insert(Mention mention);
```
Comment (kocevaIv) [inline common/src/main/java/com/brandwatch/minibcr/common/repository/MentionRepository.java]: Maybe save would be a better name?

File: common/src/main/java/com/brandwatch/minibcr/common/repository/MentionRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+import javax.annotation.PostConstruct;
+import javax.sql.DataSource;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableMap;
+
+@Repository
+public class MentionRepositoryImpl extends JdbcDaoSupport implements MentionRepository {
```
Comment (kocevaIv) [inline common/src/main/java/com/brandwatch/minibcr/common/repository/MentionRepositoryImpl.java]: Why did you decide to use JdbcDaoSupport?

File: common/src/main/java/com/brandwatch/minibcr/common/repository/MentionRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            mention = namedParameterJdbcTemplate.queryForObject(sql, new MapSqlParameterSource("mentionId", mentionId), mentionRowMapper);
+        }
+        catch (EmptyResultDataAccessException exception){
+            logger.info("Mention with id {} does not exist...", mentionId);
+        }
+
+        return mention;
+    }
+
+
```
Comment (kocevaIv) [inline common/src/main/java/com/brandwatch/minibcr/common/repository/MentionRepositoryImpl.java]: Unnecessary new line

File: api/src/main/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,12 @@
+spring.datasource.url=jdbc:postgresql://localhost:5432/minibcr
+spring.datasource.username=houssam.mahlous
+spring.datasource.password=postgres
```
Comment (kocevaIv) [inline api/src/main/resources/application.properties]: We never store the values in application.properties file and commit them to github. They should be empty, but when you work locally you can set them up yourself.

File: api/src/main/resources/application.properties [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+spring.datasource.hikari.connectionTimeout=20000
+spring.datasource.hikari.maximumPoolSize=5
+
+spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.PostgreSQL95Dialect
+spring.jpa.properties.hibernate.jdbc.lob.non_contextual_creation=true
```
Comment (kocevaIv) [inline api/src/main/resources/application.properties]: Why do you need these properties?

File: common/src/main/java/com/brandwatch/minibcr/common/domain/Mention.java [hunk 6]
```diff
@@ -10,12 +10,12 @@
 import javax.persistence.GenerationType;
 import javax.persistence.Id;
 
+@Data // Lombok: adds getters and setters
```
Comment (kocevaIv) [inline common/src/main/java/com/brandwatch/minibcr/common/domain/Mention.java]: Lombok is a useful tool, you can use it for this project. But for Brandwatch repositories we don't use it. Just giving you a heads up.

File: api/src/main/java/com/brandwatch/minibcr/api/controllers/MentionController.java [hunk 7]
```diff
@@ -17,4 +16,19 @@
     public List<Mention> getMentions(){
         return mentionService.getMentions();
     }
+
+    @GetMapping("/mentions/:id")
```
Comment (kocevaIv) [inline api/src/main/java/com/brandwatch/minibcr/api/controllers/MentionController.java]: Why does the url path have `:id`? Shouldn't it be `/mentions/{id}` instead?

File: api/src/main/java/com/brandwatch/minibcr/api/services/MentionService.java [hunk 8]
```diff
@@ -2,28 +2,50 @@
...
 
     public List<Mention> getMentions(){
         return mentionRepository.findAll();
     }
+
+    public Mention getMentionById(Long id){
+        return mentionRepository.findMentionById(id);
+    }
+
+    public void insertMention(String text) {
```
Comment (kocevaIv) [inline api/src/main/java/com/brandwatch/minibcr/api/services/MentionService.java]: saveMentions sounds better to me as a name for the method

File: common/src/main/java/com/brandwatch/minibcr/common/repository/MentionRepositoryImpl.java (hunk 2 above)
Comment (houssam-mahlous) [reply common/src/main/java/com/brandwatch/minibcr/common/repository/MentionRepositoryImpl.java]: It helped with getting a Jdbc template using `getJdbcTemplate()` without having to create a new one. However it looks unnecessary so I'll remove it

File: api/src/main/resources/application.properties (hunk 4 above)
Comment (houssam-mahlous) [reply api/src/main/resources/application.properties:3]: I will go ahead and store the values in  a local .env file instead

File: common/src/main/java/com/brandwatch/minibcr/common/domain/Mention.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
+package com.brandwatch.minibcr.common.domain;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity// This tells Hibernate to make a table out of this class
```
Comment (kocevaIv) [inline common/src/main/java/com/brandwatch/minibcr/common/domain/Mention.java]: Do you still need this annotation?

File: common/src/main/java/com/brandwatch/minibcr/common/repository/MentionRepositoryImpl.java [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+@Repository
+public class MentionRepositoryImpl implements MentionRepository {
+
+    private static final String MENTIONS_TABLE = "mentions";
+    private static final String ALL_FIELDS = " id, text";
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private final MentionRowMapper mentionRowMapper;
+    private final JdbcTemplate jdbcTemplate;
+    private final NamedParameterJdbcTemplate namedParameterJdbcTemplate;
```
Comment (kocevaIv) [inline common/src/main/java/com/brandwatch/minibcr/common/repository/MentionRepositoryImpl.java]: You don't need to use both of them. ` NamedParameterJdbcTemplate` should do the work.

**Review by kocevaIv — APPROVED** [review]
Apart from my last comments, everything else looks good to me.

**Pull request merged**